crud/src/main/java/com/redhat/lightblue/mediator/DefaultMetadataResolver.java

```diff
@@ -183,43 +185,60 @@ private void addFieldRoles(Set<String> roles, FieldTreeNode node) {
         }
     }
 
-    private void initMetadataMap(EntityMetadata emd, String entityName, String rootRequestVersion) {
-        // store root entity metadata
-        metadataMap.put(entityName, emd);
+    private void initMetadataMap(EntityMetadata emd, String rootEntityName, String rootRequestVersion) {
```

Comment:
emd is non-null at this point, a null emd will fail with null ptr exceptions later.